v2/component_analyses.yaml

```diff
@@ -0,0 +1,176 @@
+openapi: 3.0.1
+info:
+  title: Analytics Core API Specs for Version 2
+  version: "2"
+servers:
+- url: https://f8a-analytics-2445582058137.production.gw.apicast.io/api/v2
+  description: "Dependency Analytics Server"
+security:
+- tokenAuth: []
+paths:
+  /component-analyses/{eco}/{name}/{ver}:
+    get:
+      tags:
+      - Scan Services
+      summary: Get security and other details about a software component and recommendations
+        to change to a more secured version
+      operationId: da_scanner.api_v2.component-analyses
+      parameters:
+      - name: eco
+        in: path
+        description: component ecosystem
+        required: true
+        schema:
+          type: string
+      - name: name
+        in: path
+        description: component name
+        required: true
+        schema:
+          type: string
+      - name: ver
+        in: path
+        description: component version
+        required: true
+        schema:
+          type: string
+      - name: security-vendor
+        in: header
+        description: Security Vendor
```

Comment:
As we are planning to have the uuid generation at the server side, we should make sure that the component analysis too sends the uuid in its header. Initially it should send blank, and the same would be generated and sent back as the comp analysis response. The uuid can then be saved locally and sent in the subsequent calls.

If we are having a completely different endpoint (v2) then in that case do we need to pass this value?

Reply:
Added UUID. 
Regarding `security-vendor`, It is Used to identify vendor in case of Multiple vendors.
This Field is In Sync with Stack Analyses.

Reply:
`security-vendor` been removed from stack-analyses request header. Custom header is no longer as we have created a new API version to serve Snyk integration.

Reply:
@arajkumar Are you planning to change Implementations each time a new vendor Integration kicks In..?

Reply:
Of course if we drastically change the response model like in the current situation. Since we don't know how we are really going to use the field in future, it is better to avoid using `security-vendor`.

Reply:
Okay, Removing.